affected:
- package:
    ecosystem: PyPI
    name: tensorflow-gpu
    purl: pkg:pypi:tensorflow-gpu
  ranges:
  - events:
    - introduced: '0'
    - fixed: 4f663d4b8f0bec1b48da6fa091a7d29609980fa4
    repo: https://github.com/tensorflow/tensorflow
    type: GIT
  - events:
    - introduced: '0'
    - fixed: 2.2.0rc0
    - introduced: 2.2.0
    - fixed: 2.3.0rc0
    - introduced: 2.3.0
    - fixed: 2.3.4
    - introduced: 2.4.0
    - fixed: 2.4.3
    type: ECOSYSTEM
aliases:
- CVE-2021-29539
- GHSA-g4h2-gqm3-c9wq
details: TensorFlow is an end-to-end open source platform for machine learning. Calling
  `tf.raw_ops.ImmutableConst`(https://www.tensorflow.org/api_docs/python/tf/raw_ops/ImmutableConst)
  with a `dtype` of `tf.resource` or `tf.variant` results in a segfault in the implementation
  as code assumes that the tensor contents are pure scalars. We have patched the issue
  in 4f663d4b8f0bec1b48da6fa091a7d29609980fa4 and will release TensorFlow 2.5.0 containing
  the patch. TensorFlow nightly packages after this commit will also have the issue
  resolved. If using `tf.raw_ops.ImmutableConst` in code, you can prevent the segfault
  by inserting a filter for the `dtype` argument.
id: PYSEC-0000-CVE-2021-29539
modified: '2021-08-27T03:22:28.395200Z'
published: '2021-05-14T20:15:00Z'
references:
- type: ADVISORY
  url: https://github.com/tensorflow/tensorflow/security/advisories/GHSA-g4h2-gqm3-c9wq
- type: FIX
  url: https://github.com/tensorflow/tensorflow/commit/4f663d4b8f0bec1b48da6fa091a7d29609980fa4
